.circleci/config.yml

```diff
@@ -1,576 +1,20 @@
-              python_version: ["3.6", "3.7", "3.8"]
-      - build_docs
-      - docs_linkcheck
-      - all_circleci_checks_succeeded:
-          requires:
-            - e2e_tests
-            # win_e2e_tests-3.6 is not required due to the error
-            # `Cargo, the Rust package manager, is not installed or is not on PATH`.
-            # - win_e2e_tests
-            - win_e2e_tests-3.7
-            - win_e2e_tests-3.8
-            - unit_tests
-            - win_unit_tests
-            - lint
-            - pip_compile
-            - win_pip_compile
-            - build_docs
-            - docs_linkcheck
-
-  main_updated:
-    unless: <<pipeline.parameters.release_kedro>>  # don't run if 'release_kedro' flag is set
-    jobs:
-      - sync:
-          filters:
-            branches:
-              only: main
-      - viz_build:
-          filters:
-            branches:
-              only: main
-
-  hourly_pr_merge:
-    triggers:
-      - schedule:
-          cron: 0 * * * *
-          filters:
-            branches:
-              only: main
-    jobs:
-      - merge_pr_to_develop
-
-  # Python versions that are supported on `main`.
-  nightly_build_main:
-    triggers:
-      - schedule:
-          cron: 30 2 * * *
-          filters:
-            branches:
-              only: main
-    jobs:
-      - build_docker_image:
-          matrix:
-            parameters:
-              python_version: ["3.6", "3.7", "3.8"]
-      - build_kedro:
-          matrix:
-            parameters:
-              python_version: ["3.6", "3.7", "3.8"]
-          requires:
-            - build_docker_image-<<matrix.python_version>>
-
-  # Python version that are *only* supported on `develop`.
-  nightly_build_develop:
-    triggers:
-      - schedule:
-          cron: 30 2 * * *
-          filters:
-            branches:
-              only: develop
-    jobs:
-      - build_docker_image:
-          matrix:
-            parameters:
-              python_version: ["3.9"]
-      - build_kedro:
-          matrix:
-            parameters:
-              python_version: ["3.9"]
-          requires:
-            - build_docker_image-<<matrix.python_version>>
-
-  kedro_release:
-    when: <<pipeline.parameters.release_kedro>>  # only run if 'release_kedro' flag is set
+  always-run:
     jobs:
-      - build_kedro:
-          matrix:
-            parameters:
-              python_version: ["3.6", "3.7", "3.8"]
-      - publish_kedro:
-          requires:
-            - build_kedro
+      # the path-filtering/filter job determines which pipeline
+      # parameters to update, i.e. which builds to run.
+      - path-filtering/filter:
+          name: check-updated-files
+          base-revision: main
+          config-path: .circleci/continue_config.yml
+          # <regex path-to-test> <parameter-to-set> <value-of-pipeline-parameter>
+          mapping: |
+            docs/.* build-docs true
+            ^((?!docs/).)*$ build-code true
```

Comment:
I know "anything but docs" might still be too generous, but it's equivalent to what we are currently doing and we can always further refine it in the future as we progress and learn more about / pay more attention to our pain points with this. I would still like changes in `requirements.txt` for instance to still trigger the code build, but stuff like `LICENSE.md` doesn't have to. I just didn't want to spend a whole lotta time on the perfect regex.